homeassistant/components/brottsplatskartan/config_flow.py

```diff
@@ -0,0 +1,98 @@
+"""Adds config flow for Brottsplatskartan integration."""
+from __future__ import annotations
+
+from typing import Any
+import uuid
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_LATITUDE, CONF_LOCATION, CONF_LONGITUDE
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+
+from .const import AREAS, CONF_APP_ID, CONF_AREA, DEFAULT_NAME, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_LOCATION): selector.LocationSelector(
+            selector.LocationSelectorConfig(radius=False, icon="")
+        ),
+        vol.Optional(CONF_AREA, default="none"): selector.SelectSelector(
+            selector.SelectSelectorConfig(
+                options=AREAS,
+                mode=selector.SelectSelectorMode.DROPDOWN,
+                translation_key="areas",
+            )
+        ),
+    }
+)
+
+
+class BPKConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Brottsplatskartan integration."""
+
+    VERSION = 1
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a configuration from config.yaml."""
+
+        if config.get(CONF_LATITUDE):
+            config[CONF_LOCATION] = {
+                CONF_LATITUDE: config[CONF_LATITUDE],
+                CONF_LONGITUDE: config[CONF_LONGITUDE],
+            }
+        if not config.get(CONF_AREA):
+            config[CONF_AREA] = "none"
+        else:
+            config[CONF_AREA] = config[CONF_AREA][0]
+
+        return await self.async_step_user(user_input=config)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            latitude: float | None = user_input.get(CONF_LOCATION, {}).get(
+                CONF_LATITUDE
+            )
+            longitude: float | None = user_input.get(CONF_LOCATION, {}).get(
+                CONF_LONGITUDE
+            )
+            area: str | None = (
+                user_input[CONF_AREA] if user_input[CONF_AREA] != "none" else None
+            )
+
+            if area:
+                name = f"{DEFAULT_NAME} {area}"
+                unique_id = f"bpk-{area.casefold()}"
+            elif latitude and longitude:
+                name = f"{DEFAULT_NAME} {round(latitude, 2)}, {round(longitude, 2)}"
+                unique_id = f"bpk-{latitude}-{longitude}"
+            else:
+                name = f"{DEFAULT_NAME} HOME"
+                unique_id = "bpk-home"
```

Comment:
I don't think `HOME` should be a valid entry, I think it should store the current hass config location instead.
I think it should be something like this:
```suggestion
            if area:
                await self.async_abort_entry_match(....)
                title = f"{DEFAULT_NAME} {area}"
            else:
                await self.async_set_unique_id(f"bpk-{latitude}-{longitude}")
                self._abort_if_unique_id_configured()
                title = f"{DEFAULT_NAME} {round(latitude, 2)}, {round(longitude, 2)}"
```

Reply:
I would not agree with you about skipping "home" as I would think it's pretty helpful for a user to see which is for home coordinates and which is something completely different e.g. coordinates somewhere else.
Can still store the coordinates from hass config but I would still call it home (as it's also practise by many other integrations based on coordinates).

Reply:
I'm fine with calling it "home" - but IMO coordinates should still be stored.